Sources/PackageModel/PackageReference.swift

```diff
@@ -15,7 +15,9 @@ import TSCUtility
 ///
 /// This represents a reference to a package containing its identity and location.
 public struct PackageReference: Codable {
-    public typealias PackageIdentity = String
+    // FIXME: Replace with `PackageIdentity` once we change identity semantics.
+    /// The identity of the package.
+    public typealias Identity = String
```

Comment:
@mattt @neonichu @abertelrud (probably just me) but still confused why going from `String` to `PackageIdentity` should be gated on changing the identity semantics. My assumption about the goal of this PR was that we would change everywhere to use the new `PackageIdentity` type and never need to do any translations or drop down to String / legacy identity etc. Of course the actual underlying identity inside `PackageIdentity` would be String based on the same logic of extracting it from the URL, but no code other than `PackageIdentity` should know this I dont think. Then the next step would be to change how we compute the underlying (using something more unique than canonical URL) but structurally it would all works the same.

Reply:
I think we all agree that we should change the interface and the implementation separately, rather than both at the same time. So it's a question of tactics as to whether to change the interface first or the implementation first.

My PR changes the implementation while keeping the the public API mostly intact. What I believe @tomerd is suggesting is changing the interface first. Both are valid approaches that make different trade-offs.

The motivation for taking the implementation-first approach in this PR has to do with bigger interface-level changes planned for when we switch over the semantics. For example, 960642b830bf0ceceac3306534d02aca1a52f367 changes `PackageReference` from being a structure to an enumeration to better model the distinction between local and remote package dependencies. By starting with the implementation, I believe we'll minimize risk and the be able to get to where we want to go in two PRs. If we started with the interface, I think we'd have to split out the other changes into a third, separate PR, or risk biting off more than we can chew at once in just one.